docs/spanner.md

```diff
@@ -0,0 +1,168 @@
+
+## Steps to use Spanner for your cluster
+
+All commands below should be run from the root directory of this project.
+Run the commands in the same session to keep environmental variables consistent.
+
+### 0. Environmental Variables
+
+Choose your desired project, region, and zone.
+
+```sh
+PROJECT_ID="<your-project-id>"
+REGION="<your-GCP-region>"      # e.g. us-central1
+ZONE="<your-GCP-zone>"          # e.g. us-central1-b
+```
+
+### 1. Create a GKE cluster with Workload Identity enabled
+
+These commands are similar as what is listed in the README Quickstart (GKE),
+but include a flag to [enable workload identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable).
+This will allow the GKE service account to query Spanner.
+
+GKE autopilot mode (see [Autopilot
+overview](https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-overview)
+to learn more):
+
+```sh
+gcloud container clusters create-auto onlineboutique \
+    --project=${PROJECT_ID} --region=${REGION}
+```
+
+GKE Standard mode:
+
+```sh
+gcloud container clusters create onlineboutique \
+    --project=${PROJECT_ID} --zone=${ZONE} \
+    --workload-pool=${PROJECT_ID}.svc.id.goog \
+    --machine-type=n2-standard-2 --num-nodes=4
+```
+
+### 2. Enable Cloud Spanner on your project
+
+```sh
+gcloud services enable spanner.googleapis.com --project=${PROJECT_ID}
+```
+
+### 3. Create a Cloud Spanner instance, database, and table
+
+Note: See the documentation to list [available Spanner configuration names](https://cloud.google.com/spanner/docs/getting-started/set-up#run_the_gcloud_tool), or run `gcloud spanner instance-configs list`
+
+```sh
+SPANNER_REGION_CONFIG="<your-spanner-region-config-name>"
+# e.g. "regional-us-central1"
+
+gcloud spanner instances create hipstershop \
+    --description="online boutique backend" \
+    --config="${SPANNER_REGION_CONFIG}" \
+    --instance-type=free-instance
+
+gcloud spanner databases create carts \
+    --project=${PROJECT_ID} \
+    --instance="hipstershop" \
+    --database-dialect=GOOGLE_STANDARD_SQL \
+    --ddl-file=./src/cartservice/ddl/CartItems.ddl
+```
+
+### 4. Grant the default GCE/GKE service account read/write permission to Spanner
+
+```sh
+# grant the default GKE service account databaseUser in Spanner
+SVCACCT=$(gcloud iam service-accounts list --filter 'Compute Engine default' --format 'value(email)')
+gcloud projects add-iam-policy-binding $PROJECT_ID \
+    --member "serviceAccount:${SVCACCT}" \
+    --role="roles/spanner.databaseUser"
+
+# annotate the kubectl serviceAccountName with the gcloud svcaccount
+kubectl annotate serviceaccount default \
+    iam.gke.io/gcp-service-account=${SVCACCT}
+
+# tell gcloud that the kubectl account maps to the gcloud one
+gcloud iam service-accounts add-iam-policy-binding \
+    "${SVCACCT}" --role roles/iam.workloadIdentityUser \
+    --member "serviceAccount:${PROJECT_ID}.svc.id.goog[default/default]"
```

Comment:
Could you use variable for the 2 `default` values (`NAMESPACE` versus `SERVICE_ACCOUNT`, they are used at different places on that `.md` file and that would help for example the end-users who would like to deploy Online Boutique in another namespace, etc.

Reply:
I think this should cover it: 46b9ee27217f328f40d771f7dc43a50f39dba2ed and b0069332f511dd336fef0a8392b491abdc0a3c17, please let me know.

Reply:
Also, just aligned the variable names with those in `workload-identity.md`: f5097037783d9bb953a3ba7a60f7e09a1fe8690b